src/picongpu/include/plugins/ResourceLog.hpp

```diff
@@ -0,0 +1,244 @@
+                pt.put("resourceLog.rank", rank);
+            }
+
+            if(contains(propertyMap,"position"))
+            {
+                DataSpace<simDim> currentPosition = Environment<simDim>::get().GridController().getPosition();
+                pt.put("resourceLog.position.x", currentPosition[0]);
+                pt.put("resourceLog.position.y", currentPosition[1]);
+                pt.put("resourceLog.position.z", currentPosition[2]);
+            }
+
+            if(contains(propertyMap, "currentStep"))
+            {
+                pt.put("resourceLog.currentStep", currentStep);
+            }
+
+            if(contains(propertyMap, "cellCount"))
+            {
+                size_t cellCount = resourceMonitor.getCellCount();
+                pt.put("resourceLog.cellCount", cellCount);
+            }
+
+            if(contains(propertyMap,"particleCount"))
+            {
+                std::vector<size_t> particleCounts = resourceMonitor.getParticleCounts<VectorAllSpecies>(*cellDescription);
+                pt.put("resourceLog.particleCount", std::accumulate(particleCounts.begin(), particleCounts.end(), 0));
+            }
+
+            //
+            // Write property tree to string stream
+            std::stringstream ss;
+            if(outputFormat == "json")
+            {
+                write_json(ss, pt, false);
+            }
+            else if(outputFormat == "jsonpp")
+            {
+                write_json(ss, pt, true);
+            }
+            else if(outputFormat == "xml")
+            {
+                write_xml(ss, pt);
+            }
+            else if(outputFormat == "xmlpp")
+            {
+                write_xml(ss, pt, boost::property_tree::xml_writer_make_settings<std::string>('\t', 1));
+            }
+            else
+            {
+                throw std::runtime_error(std::string("resourcelog.format ") + outputFormat + std::string(" is not known, use json or xml."));
+            }
+
+            //
+            // Write property tree to the output stream
+            if(streamType == "stdout")
+            {
+                std::cout << ss.str();
+            }
+            else if (streamType == "stderr")
+            {
+                std::cerr << ss.str();
+            }
+            else if (streamType == "file")
+            {
+                std::ostream os(&fileBuf);
+                os << ss.str();
+            }
+            else
+            {
+                throw std::runtime_error(std::string("resourcelog.stream ") + streamType + std::string(" is not known, use stdout, stderr or file instead."));
+            }
+        }
+
+        void pluginRegisterHelp(po::options_description& desc)
+        {
+            /* register command line parameters for your plugin */
+            desc.add_options()
+                    ("resourceLog.period", po::value<uint32_t>(&notifyPeriod)->default_value(0),
```

Comment:
please add an example to `doc/TBG_macros.cfg` and document the plugin in our [wiki](https://github.com/ComputationalRadiationPhysics/picongpu/wiki/PIConGPU-Plugins) :)